service/storage_service.cc

```diff
@@ -814,6 +814,26 @@ void storage_service::join_token_ring(int delay) {
         // If we're replacing, _cdc_streams_ts should have been set in prepare_to_join.
         assert(!db().local().is_replacing());
 
+        if (!db::system_keyspace::bootstrap_complete() /* second case */) {
+            // Check if we managed to learn about other nodes' tokens (if there are other nodes).
+            // We might have not due to misconfiguration, e.g. skipping the "wait for gossip to settle" phase,
+            // or due to a network partition which didn't allow us to contact other seeds.
+            // But CDC requires that we know other nodes' tokens - they are needed to make a new generation of streams.
+            const auto& seeds = _gossiper.get_seeds();
+            if (!seeds.count(get_broadcast_address()) || seeds.size() > 1) {
+                auto& tok_to_ep = _token_metadata.get_token_to_endpoint();
+                if (std::none_of(tok_to_ep.begin(), tok_to_ep.end(),
+                        [me = get_broadcast_address()] (const std::pair<token, inet_address> p) { return p.second != me; })) {
+                    throw std::runtime_error(format(
+                        "Can't boot this node because we weren't able to retrieve other nodes' tokens,"
+                        " which is required for CDC to work. Make sure that:\n"
+                        " - you don't use the \"--skip-wait-for-gossip-to-settle\" flag,\n"
```

Comment:
This is a problem. Basically, you say the option "--skip-wait-for-gossip-to-settle" must not be used, or you will likely see this error? If this is the case, shouldn't we remove this option? Does it still have a legitimate use case?

Reply:
There is no harm in using this option e.g. when starting the first node, because it doesn't need to gossip with anyone, and it speeds up the first node's startup significantly.

I don't know other legitimate uses though. Maybe if the user wants to start doing writes immediately even if they will go to wrong replicas, and then run repair, or something. So the cluster starts faster.